### PR TITLE
CommuniKate for all languages and Presage contraction support

### PR DESCRIPTION
**Pull request by zelmon64**
The CommuniKate keyboard can now be enabled and used by any language. To access it a key has been added to the Output section.

Apostrophe support has been added to Presage along with the database from ACAT which contains contractions.

### CODE REVIEW

File: src/JuliusSweetland.OptiKey/UI/ViewModels/Management/WordsViewModel.cs [hunk 1]
```diff
@@ -147,7 +147,8 @@ public bool UseDefaultKeyboardLayoutIsVisible
                        || KeyboardAndDictionaryLanguage == Enums.Languages.EnglishUK
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/UI/ViewModels/Management/WordsViewModel.cs]: Could you please change this to instead check UseAlphabeticalKeyboardLayoutIsVisible || UseSimplifiedKeyboardLayoutIsVisible || EnableCommuniKateKeyboardLayoutIsVisible

File: src/JuliusSweetland.OptiKey/UI/ViewModels/Management/WordsViewModel.cs [hunk 2]
```diff
@@ -111,7 +111,7 @@ public Languages KeyboardAndDictionaryLanguage
                 SetProperty(ref this.keyboardAndDictionaryLanguage, value);
                 OnPropertyChanged(() => UseDefaultKeyboardLayoutIsVisible);
                 OnPropertyChanged(() => UseAlphabeticalKeyboardLayoutIsVisible);
-                OnPropertyChanged(() => EnableCommuniKateKeyboardLayoutIsVisible);
+                OnPropertyChanged(() => EnableCommuniKateKeyboardLayout);
```
Comment (JuliusSweetland) [inline src/JuliusSweetland.OptiKey/UI/ViewModels/Management/WordsViewModel.cs]: Shouldn't this be UseCommuniKateKeyboardLayoutIsVisible?

**Comment by JuliusSweetland** [comment]
@zelmon64 Hi William - this looks good, but I'm at work so I can't test for a couple of days. I've added one very minor comment - if you have time to change it that would be great, otherwise I'll change it when I test. Thanks again!

**Comment by JuliusSweetland** [comment]
@zelmon64 Few questions:

1.How does the presage database setting in OptiKey (in the Management Console) work with the presage xml config value, which I think is this one:
<Predictors>
        <DefaultSmoothedNgramPredictor>
            <PREDICTOR>SmoothedNgramPredictor</PREDICTOR>
            <LOGGER>ERROR</LOGGER>
            <DBFILENAME>::PACKAGEDATADIR::/database_en.db</DBFILENAME>

??? Does OptiKey's setting (if set) override the presage setting?

2.Do I need to update this page to change the customisation of the xml file in step 5? https://github.com/OptiKey/OptiKey/wiki/Next-word-prediction

3.Do I need to also change that page to change how the user changes the presage language (bottom of page https://github.com/OptiKey/OptiKey/wiki/Next-word-prediction)

**Comment by zelmon64** [comment]
@JuliusSweetland 

1. The new settings get saved to `C:\Users\user\.presage\presage.xml` and override the default presage settings.
2. Step 5 can be removed. Those settings are now set automatically.
3. Yes, to use a different language database it must now be set from the Management Console.

**Pull request merged**